src/librustdoc/json/conversions.rs

```diff
@@ -107,7 +108,7 @@ impl JsonRenderer<'_> {
             Some(ty::Visibility::Public) => Visibility::Public,
             Some(ty::Visibility::Restricted(did)) if did.is_crate_root() => Visibility::Crate,
             Some(ty::Visibility::Restricted(did)) => Visibility::Restricted {
-                parent: from_item_id(did.into(), self.tcx),
+                parent: id_from_item_inner(did.into(), self.tcx, None, None),
```

Comment:
This seems like a pretty common pattern. I think it's worth writing a helper function, instead of writing `None, None` everywhere .